sdk/dotnet/Pulumi.Automation/Exceptions/ProjectSettingsConflictException.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright 2016-2021, Pulumi Corporation
+
+using System;
+
+namespace Pulumi.Automation.Exceptions
+{
+    /// <summary>
+    ///
+    /// Thrown when creating a Workspace detects a conflict between
+    /// project settings found on disk (such as Pulumi.yaml) and a
+    /// ProjectSettings object passed to the Create API.
+    ///
+    /// There are two resolutions:
+    ///
+    /// (A) to use the ProjectSettings, delete the Pulumi.yaml file
+    ///     from WorkDir or use a different WorkDir
+    ///
+    /// (B) to use the exiting Pulumi.yaml from WorkDir, avoid
+    ///     customizing the ProjectSettings
+    ///
+    /// </summary>
+    public class ProjectSettingsConflictException : Exception
+    {
+        internal string SettingsFileLocation { get; }
+
+        internal ProjectSettingsConflictException(string settingsFileLocation)
+            : base(ErrMsg(settingsFileLocation))
```

Comment:
You can do the string interpolation right here instead of having a single-use method:
```csharp
internal ProjectSettingsConflictException(string settingsFileLocation)
    : base($"Custom {nameof(ProjectSettings)} passed in code conflict with settings found on disk: {settingsFileLocation}")
{
    SettingsFileLocation = settingsFileLocation;
}
```

I like using the `nameof(...)` whenever I can when referring to a class in the library, which helps to enforce that string literals get updated whenever a class name changes. I try to do the same thing in xml comments using the `<see ...>` tag whenever referring to a class.